parlai/core/teachers.py

```diff
@@ -242,6 +239,13 @@ def num_episodes(self):
         except Exception:
             raise RuntimeError('"num_episodes" must be overriden by children.')
 
+    def num_examples(self):
```

Comment:
so what about if num_episodes (or num_examples) isnt defined -- what happens?

Reply:
Currently, any subclass of `FixedDialogTeacher` that does not have `self.episodes` and `self.examples`, and does not override `num_episodes` or `num_examples`, will get a runtime error. However, `DialogTeacher` has these functions implemented, and all of the teachers that subclass `FixedDialogTeacher` (in this repo) have these functions implemented as well

Reply:
i think the base class teacher should have them defined with returning whatever means 'unknown'  (for e.g. streaming or infinite data where we dont want to specify)

Reply:
Ok, do you mean like the base `Teacher` class? Before, we had a return of 0 be the default for streaming data in the `StreamDialogData`, but now we compute the length once and then save that to file

Reply:
i mean for tasks where the data is infinite, or where they can't/don't declare it.  we probably shouldn't use 0 because it's numerical -- somehow None or inf seem to make more sense?

Reply:
I think None makes sense; I can add that to the base `Teacher` class, if you think that's the right call

Reply:
yep cool,  then anything that sums over num_*  that sees a None in one of the worlds, should also return None, right?

Reply:
Yeah, I'll make that change right now

Reply:
(also--the reason we used 0 as the default is that __len__ is only allowed to return positive numbers or zero, can't do -1 or None or anything. None is good!)